src/sentry/workflow_engine/migration_helpers/alert_rule.py

```diff
@@ -529,3 +583,45 @@ def dual_delete_migrated_alert_rule(
         RegionScheduledDeletion.schedule(instance=data_source, days=0, actor=user)
 
     return
+
+
+def dual_delete_migrated_alert_rule_trigger(
+    alert_rule_trigger: AlertRuleTrigger,
+    user: RpcUser | None = None,
+) -> None:
+    priority = (
+        DetectorPriorityLevel.MEDIUM
+        if alert_rule_trigger.label == "warning"
+        else DetectorPriorityLevel.HIGH
+    )
+    detector_trigger = get_detector_trigger(alert_rule_trigger, priority)
+    if detector_trigger is None:
+        return None
+
+    action_filter = get_action_filter(alert_rule_trigger, priority)
+    if action_filter is None:
+        return None
+
+    detector_trigger.delete()
+    action_filter.delete()
+    logger.info("Dual deleted alert_rule_trigger", extra={"alert_rule_trigger": alert_rule_trigger})
+
+    return None
+
+
+def dual_delete_migrated_alert_rule_trigger_action(
+    trigger_action: AlertRuleTriggerAction,
+    user: RpcUser | None = None,
+) -> None:
+    aarta = ActionAlertRuleTriggerAction.objects.filter(
+        alert_rule_trigger_action=trigger_action
+    ).first()
+    if aarta is None:
+        logger.error(
```

Comment:
thoughts on throwing an error here instead of logging an error? doing that might help flow control, we could also update the typing to always have a return statement, and if we hit one of these unexpected cases we throw an error to loudly say there's an issue here.

doing that, also means we can handle these cases differently for any invocation of this method.